cddl_gen/cddl_gen.py

```diff
@@ -269,8 +275,15 @@ def set_base_name(self, base_name):
 
     # Add uniqueness to the base name.
     def id(self):
-        return "%s%s" % ((self.id_prefix + "_")
-                         if self.id_prefix != "" else "", self.get_base_name())
+        raw_name = self.get_base_name()
+        if self.id_prefix \
+                and (f"{self.id_prefix}_" not in raw_name) \
+                and (self.id_prefix != raw_name.strip("_")):
```

Comment:
```suggestion
        if (self.id_prefix
            and (f"{self.id_prefix}_" not in raw_name)
            and (self.id_prefix != raw_name.strip("_"))):
```
Backslashes give me the impression that the first logic chunk itself can't fit on one line, not just separating the logic statements comprising the if.